File: src/main/java/ac/kr/deu/connect/luck/auth/dto/SignUpRequest.java
package ac.kr.deu.connect.luck.auth.dto;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "회원가입 요청 정보 객체")
public record SignUpRequest(
        @Schema(description = "이메일", example = "test@test.com")
        String email,
        @Schema(description = "비밀번호", example = "test")
        String password,
        @Schema(description = "이름", example = "홍길동")
        String name,
        @Schema(description = "휴대폰 번호", example = "010-1234-5678")
        String phoneNumber
) {
}

File: src/main/kotlin/ac/kr/deu/connect/luck/auth/controller/request/SignupRequest.kt
package ac.kr.deu.connect.luck.auth.controller.request

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Pattern

@Schema(description = "회원가입 요청 폼")
class SignupRequest(
    @field:Email
    @field:Schema(description = "이메일", example = "test@example.com")
    val email: String,
    @field:Schema(description = "비밀번호", example = "password")
    val password: String,
    @field:Schema(description = "이름", example = "홍길동")
    val name: String,
    @field:Pattern(regexp = "^\\d{3}-\\d{3,4}-\\d{4}\$")
    @field:Schema(description = "전화번호", example = "010-1234-5678")
    val phoneNumber: String,
)
